types/eslint/index.d.ts

```diff
@@ -1178,8 +1166,8 @@ export namespace Linter {
     }
 
     // Temporarily loosen type for just flat config files (see #68232)
-    type FlatConfigParserModule =
-        & Omit<ParserModule, "parseForESLint">
+    type NonESTreeParser =
+        & Omit<ESTreeParser, "parseForESLint">
```

Comment:
[Naming] Would this not be a Non-_ESLint_-Parser then?

Reply:
Looking down a few lines you can see that this type includes a `parseForESLint` method, but with a different signature than the `ESTreeParser` type. The method is omitted only to be redefined with a different type. So I'd say it's still an ESLint-specific parser.